aiida/orm/nodes/data/remote/base.py

```diff
@@ -149,18 +149,22 @@ def listdir_withattributes(self, path='.'):
                 else:
                     raise
 
-    def _clean(self):
+    def _clean(self, transport=None):
         """
         Remove all content of the remote folder on the remote computer
         """
         from aiida.orm.utils.remote import clean_remote
 
-        authinfo = self.get_authinfo()
-        transport = authinfo.get_transport()
         remote_dir = self.get_remote_path()
+        if transport is None:
+            authinfo = self.get_authinfo()
+            transport = authinfo.get_transport()
 
-        with transport:
+            with transport:
+                clean_remote(transport, remote_dir)
+        else:
             clean_remote(transport, remote_dir)
+        self.set_extra('cleaned', True)
```

Comment:
Would make the `'cleaned'` constant a class attribute of `RemoteData`, e.g., `KEY_EXTRA_CLEANED = 'cleaned'`. Then you can use this attribute in `get_calcjob_remote_paths` to do
```
filters_remote[f'extras.{RemoteData.KEY_EXTRA_CLEANED}'] = {'!==': True}
```

Reply:
good point